File: CSharp/OpenTraceability/Utility/Attributes/OpenTraceabilityMasterDataAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace OpenTraceability.Utility.Attributes
{
    /// <summary>
    /// This attribute indicates how the property is mapped into the EPCIS Master Data header attributes.
    /// </summary>
    public class OpenTraceabilityMasterDataAttribute : Attribute
    {
        public string Name { get; set; }

        public OpenTraceabilityMasterDataAttribute()
        {
            Name = string.Empty;
        }

        public OpenTraceabilityMasterDataAttribute(string name)
        {
            Name = name;
        }

        public OpenTraceabilityMasterDataAttribute(string ns, string name)
        {
            Name = (((XNamespace)ns) + name).ToString();
        }
    }
}

File: Java/OpenTraceability/src/main/opentraceability/utility/attributes/opentraceabilitymasterdataattribute.kt
package utility.attributes

import java.lang.reflect.Type

class OpenTraceabilityMasterDataAttribute /*: Attribute*/ {
    var Name: String = ""

    constructor() {
        Name = ""
    }
    constructor(name: String) {
        Name = name
    }
    constructor(ns: String, name: String) {
        //Name = (((XNamespace)ns) + name).ToString();
        Name = name
    }
}
